### PR TITLE
[SYCL][FE][Driver] Re-enable fp-accuracy tests

### PR DESCRIPTION
**Pull request by zahiraam**
Both issues were due to the DoRoundTrip value (1 in debug compile and 0 in release compile). 
Diagnostics were generated in a function dependent on the value of DoRoundTrip and that had the effect of generating no diagnostic in release mode, but diagnostics were fired in debug more in the second run on the function RoundTrip.
Same for the setting of the codegen’s FPAccuracy option.
Changes:
1- Removed un-necessary FPaccuracy codegen option; used instead the values of FpAccuracyVal and FPAccuracyFuncMap.
2- Changed place to the diagnostics. 
@AaronBallman   what do you think of the placement of the diagnostics? Is this acceptable for potential upstream?

Fixes: https://github.com/intel/llvm/issues/9934

### CODE REVIEW

**Review by mdtoguchi — APPROVED** [review]
OK for driver

**Comment by dm-vodopyanov** [comment]
@zahiraam please, follow https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#commit-message

**Review by premanandrao — APPROVED** [review]
FE changes look okay to me.   Waiting for @AaronBallman's comments on what @zahiraam asked.

File: clang/lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -4700,6 +4687,14 @@ bool CompilerInvocation::CreateFromArgs(CompilerInvocation &Invocation,
         Invocation.generateCC1CommandLine(Args, SA);
       },
       Invocation, DummyInvocation, CommandLineArgs, Diags, Argv0);
+
+  // Diagnose FPAccuracy option validity.
+  LangOptions &LangOpts = *Invocation.getLangOpts();
```
Comment (AaronBallman) [inline clang/lib/Frontend/CompilerInvocation.cpp:4692]: This doesn't look correct to me -- option round-tripping shouldn't be anything special like this.

I think the original issue was that you were doing work in `ParseFpAccuracyArgs()` which is called from `ParseLangArgs()` but not updating `GenerateLangArgs()` to call `GenerateArg()` for the argument you're parsing.

**Review by AaronBallman — APPROVED** [review]
LGTM!

**Comment by zahiraam** [comment]
@intel/llvm-gatekeepers can this be merged? Thanks.

**Pull request merged**